src/MoveToHelper.cc

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "ignition/rendering/MoveToHelper.hh"
+
+#include <memory>
+
+#include <ignition/common/Animation.hh>
+#include <ignition/common/KeyFrame.hh>
+#include <ignition/math/Pose3.hh>
+
+#include "ignition/rendering/Camera.hh"
+
+class ignition::rendering::MoveToHelperPrivate
+{
+  /// \brief Pose animation object
+  public: std::unique_ptr<common::PoseAnimation> poseAnim;
+
+  /// \brief Pointer to the camera being moved
+  public: rendering::CameraPtr camera;
+
+  /// \brief Callback function when animation is complete.
+  public: std::function<void()> onAnimationComplete;
+
+  /// \brief Initial pose of the camera used for view angles
+  public: math::Pose3d initCameraPose;
+};
+
+using namespace ignition;
+using namespace rendering;
+
+MoveToHelper::MoveToHelper() :
+  dataPtr(new MoveToHelperPrivate)
```

Comment:
Nit: `new` isn't needed here.

Reply:
are you sure ? dataPtr is a `unique_ptr`.

Reply:
Oh yes, sorry.

Reply:
It's best to use `make_unique`